### PR TITLE
refactor(dart/transform): Remove index_static from hello_world

### LINKED ISSUE #495: Remove index_static.js from hello_world example.
Once the dart transformers work, we should be able to remove [`index_static.js`](https://github.com/angular/angular/blob/master/modules/examples/src/hello_world/index_static.js) from the source code since transformers will now be generating the file.

Requires issue #: 
 - [x] #496 Transformers create constructor stubs
 - [ ] #497 Tree shake constructor stubs
 - [x] #498 Transformers include annotations for each constructor stub
 - [x] #500 Generate setter stubs for the Directives
 - [ ] #501 Parse HTML templates to extract stubs from expressions & statements, then generate getters/setters stubs for reflection
 - [x] #503 Mock out DOM API so that compiler can execute outside the browser
 - [ ] #502 Generate Change Detection classes instead of getter setter stubs from #501, #503.
 - [x] #700 Remove import of dart:mirrors

### PR DESCRIPTION
**Pull request by kegluneq**
index_static.js & index_static.html are unnecessary in Js and are now
essentially generated via the Dart transformer. The `angular`
transformer is specified in `examples/pubspec.yaml`; use `pub build` to
create a transformed application that does not use dart:mirrors.

Closes #495

### CODE REVIEW

File: modules/examples/src/hello_world/index.js [hunk 1]
```diff
@@ -1,14 +1,96 @@
...
+// A service available to the Injector, used by the HelloCmp component.
+@Injectable()
+class GreetingService {
+  greeting:string;
+  constructor() {
+    this.greeting = 'hello';
+  }
+}
+
+function createBindings() {
```
Comment (kegluneq) [inline modules/examples/src/hello_world/index.js]: Think this is worth adding? I believe it runs fine without, but this is added for parity with `index_static.js`.

File: modules/examples/src/hello_world/index.js [hunk 2]
```diff
@@ -1,14 +1,96 @@
...
+}
 
 export function main() {
   // Initializing the reflector is only required for the Dart version of the application.
   // When using Dart, the reflection information is not embedded by default in the source code
   // to keep the size of the generated file small. Importing ReflectionCapabilities and initializing
   // the reflector is required to use the reflection information from Dart mirrors.
-  // Dart mirrors are not intended to be use in production code where the transformers generate a
-  // more optimal static configuration, see index_static.js for an example.
+  // Dart mirrors are not intended to be use in production code.
```
Comment (kegluneq) [inline modules/examples/src/hello_world/index.js]: What do you think of this wording? Think we need a link to [Pub Assets and Transformers][] somewhere around here? Should we more explicitly call out that you can ignore this if you're using Js?

[Pub Assets and Transformers]: https://www.dartlang.org/tools/pub/assets-and-transformers.html

**Comment by kevmoo** [comment]
Woo hoo! :metal:

File: modules/examples/src/hello_world/index.js (hunk 2 above)
Comment (yjbanov) [inline modules/examples/src/hello_world/index.js]: Should be ok if you start the line with "For Dart users:". Also mention that the transformer _replaces_ the code below, not just generates something.

File: modules/examples/src/hello_world/index.js (hunk 1 above)
Comment (yjbanov) [inline modules/examples/src/hello_world/index.js]: Probably not worth adding for hello world. It should be as simple as possible.

**Comment by kegluneq** [comment]
I'm going to put this on hold briefly -- looking at the Travis error pointed me towards the e2e tests. It looks like they're trying to test that things work both reflectively & statically, but now that we're using the transformer we're no longer testing the reflective case.

I'm going to put together another PR to address that situation and then I'll come back to this.

**Comment by kegluneq** [comment]
I think https://github.com/angular/angular/pull/1436 is a better way to do this, although it ends up with an additional `index_static.js` & `index_static.html` rather than just having the two files in the dir.

Makes reflective vs. static testing a lot easier though.

**Comment by yjbanov** [comment]
SGTM. Let's close this one then.

**Pull request closed without merging**